src/urllib3/response.py

```diff
@@ -208,20 +389,19 @@ def __init__(
         request_url=None,
         auto_close=True,
     ):
+        super().__init__(
+            headers=headers,
+            status=status,
+            version=version,
+            reason=reason,
+            decode_content=decode_content,
+            enforce_content_length=enforce_content_length,
+        )
 
-        if isinstance(headers, HTTPHeaderDict):
-            self.headers = headers
-        else:
-            self.headers = HTTPHeaderDict(headers)
-        self.status = status
-        self.version = version
-        self.reason = reason
-        self.decode_content = decode_content
         self.retries = retries
         self.enforce_content_length = enforce_content_length
```

Comment:
This line can be removed as it's already in `BaseHTTPResposne`

Reply:
I'm still seeing that extra line, is that expected?

Reply:
@pquentin I think I moved the `enforce_content_length` back into `HTTPResponse`, are things okay now with the diff?

Reply:
Ahh right I hadn't realized that.

To be honest this is why I don't like rebases during pull requests. Since GitHub doesn't have a way to show what changed during two pushes (unlike GitLab and tools like Gerrit), rebasing is complicating the work of the reviewer who has to read the whole diff again and can easily miss things like this.

I'd suggest just adding commits, which is fine since we squash commits on merge anyway.

Reply:
Sorry for the confusion, will do in the future.